daffodil-runtime1/src/main/scala/org/apache/daffodil/udf/UserDefinedFunctionService.scala

```diff
@@ -0,0 +1,185 @@
+   * the evaluate's parameter types and evaluate's return type
+   */
+  def lookupUDFCallingInfo(namespaceURI: String, fname: String): Option[UdfCallingInfo] = {
+    val udfid = s"$namespaceURI:$fname"
+
+    val udfFunctionCallingInfo = InitializedUserDefinedFunctionsCache.getOrElse(udfid, {
+      val provider = functionProviderLookup.get(udfid)
+      val fClass = functionClassesLookup.get(udfid)
+      if (!List(provider, fClass).exists(_.isEmpty)) {
+        val udf = provider.get.lookupInitializedUserDefinedFunction(namespaceURI, fname)
+        val expectedUdfClass = fClass.get
+
+        if (udf != null) {
+          val actualUdfClass = udf.getClass
+          if (actualUdfClass != expectedUdfClass) {
+            log(LogLevel.Error, s"User Defined Function Class Mismatch: $udfid. Expected: $expectedUdfClass Actual: $actualUdfClass")
+            None
+          } else {
+
+            val evaluateMethodInfoTuples = expectedUdfClass.getMethods.collect {
+              case p if p.getName == evaluateMethodName => (p, p.getParameterTypes, p.getReturnType)
+            }
+
+            if (evaluateMethodInfoTuples.isEmpty) {
+              log(LogLevel.Error, s"Missing evaluate method for function provided: $udfid")
+              None
+            } else if (evaluateMethodInfoTuples.length > 1) {
+              log(LogLevel.Error, s"Only one evaluate method allowed per function class: $udfid")
+              None
+            } else if (List(classOf[java.lang.Void], classOf[Unit]).exists(_ == evaluateMethodInfoTuples.head._3)) {
+              log(LogLevel.Error, s"Evaluate method for function provided cannot be void: $udfid")
+              None
+            } else {
+              val evaluateMethodInfo: (EvaluateMethod, ParameterTypes, ReturnType) = evaluateMethodInfoTuples.head
+              /*
+               * Attempting assignnment by tuple matching results in a scala existentials error in SBT
+               * so we do it this way instead
+               */
+              val evaluateMethod: EvaluateMethod = evaluateMethodInfo._1
+              val paramTypes: ParameterTypes = evaluateMethodInfo._2
+              val retType: ReturnType = evaluateMethodInfo._3
+              val udfci: Option[UdfCallingInfo] = Some((udf, evaluateMethod, paramTypes, retType))
+              InitializedUserDefinedFunctionsCache += (udfid -> udfci)
+              udfci
+            }
+          }
+        } else {
+          log(LogLevel.Error, s"Function not found: $udfid")
+          log(LogLevel.Info, s"Currently registered UDFs:\n${UserDefinedFunctionService.allFunctionClasses}")
+          None
+        }
+      } else {
+        val requiredThing = List(
+          (if (provider.isEmpty) "Provider" else ""),
+          (if (fClass.isEmpty) "User Defined Function" else ""))
+          .filter(s => !Option(s).isEmpty && s.trim.nonEmpty).mkString("/")
+        log(LogLevel.Error, s"$requiredThing not found: $udfid")
+        None
```

Comment:
Not sure I woud log this. The caller will likely cause an SDE about "uknown function" and there is enough logging above that should give a clue as to why it wasn't found.